client/verta/verta/tracking/entities/_deployable_entity.py

```diff
@@ -88,6 +90,74 @@ def _get_artifact_msg(self, key):
     def _get_artifact(self, key):
         raise NotImplementedError
 
+    def set_schema(self, input: dict, output: dict = None) -> None:
+        """
+        Sets the input and output schemas of this Deployable Entity. Schemas are stored as
+        model artifacts.
+
+        The output schema is optional.
+
+        To validate a prediction's input and output against these schemas, use the
+        :func:`~verta.registry.validate_schema` decorator.
+
+        Parameters
+        ----------
+        input : dict
+            Input schema as an OpenAPI-compatible json dict. Easiest to create using pydantic.BaseModel.schema() [#]_.
+        output : dict
+            Output schema as an OpenAPI-compatible json dict. Easiest to create using pydantic.BaseModel.schema().
+
+
+        References
+        ----------
+        .. [#] https://docs.pydantic.dev/1.10/usage/schema/
+
+        """
+        if not isinstance(input, dict):
+            raise TypeError(
+                f"`input` must be of type dict, not {type(input)}; did you remember to call `.schema()`?"
+            )
+        if output is not None and not isinstance(output, dict):
+            raise TypeError(
+                f"`output` must be of type dict, not {type(output)}; did you remember to call `.schema()`?"
+            )
+
+        schema = {
+            "input": input,
+        }
+        if output is not None:
+            schema["output"] = output
+
+        # write a temp file because otherwise `log_artifact` will think the artifact contents are the file path
+        with tempfile.NamedTemporaryFile(mode="w") as temp_file:
+            temp_filename = temp_file.name
+            with open(temp_filename, "w") as file:
+                json.dump(schema, file)
+            self.log_artifact(
+                key="model_schema_json", artifact=temp_filename, overwrite=True
```

Comment:
I'd recommend

1. Above, using `tempfile.NamedTemporaryFile(..., suffix=".json")` which will have the client log the artifact as a JSON file
   ```
   filename_extension: "json"
   ```
   and allow the web UI to recognize and download the artifact as JSON
   ![Screenshot 2023-07-13 at 10 00 07 AM](https://github.com/VertaAI/modeldb/assets/96442646/bf31023d-45d1-4d7d-9457-3bf63f12307f)
2. Making the artifact key `model_schema` because it's redundant, but also in general I don't think we need `json` in the key anyway 😬 
   ```suggestion
                   key="model_schema", artifact=temp_file.name, overwrite=True
   ```
3. Maybe `overwrite` should be a parameter on this method 🤔

Reply:
3. if this function is supposed to update a model_schema, I don't see why we'd ever not want to overwrite it?